### PR TITLE
Solving a one way ANOVA with Excel and testing for normality and homogeneity using Python

### LINKED ISSUE #5017: Solving a one way anova with Excel and testing for normality and homogeneity using python
#### Solving a one way ANOVA with Excel and testing for normality and homogeneity using Python 
Your title should be descriptive of the article/tutorial. Be Specific. 

Use keyword research to improve your article's SEO.

#### Proposed article introduction
Over the years, data has been an essential part of how the world works. These data range from GDP to blood samples to every aspect of the world. As data grew, statistics found ways to make meaning of them. One of them is the analysis of variance. ANOVA is a collection of statistical models and their associated estimation procedures to analyse the differences among means.

As we entered more technologically advanced stages, tools were created for ease in analysing this data. For ANOVA, Microsoft came up with statistical add-ins within Excel to solve some of these problems. Though the effort, Microsoft Excel was still limited. There came tools like R-studio and Python. With Python, the test for normality and homogeneity became easy. 

This tutorial will focus on box plots to test the normality of a one way ANOVA and Bartlett's rule for homogeneity with Python.

#### Key takeaways 
At the end of the tutorial, the reader will be able to:
- Use Excel to create an ANOVA table. 
- Install necessary Python dependencies to create a box plot 
- Test for normality using a box plot 
- Test for homogeneity using Bartlett's rule
### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Mirror98**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5017

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Mirror98** [comment]
@hectorkambow @Kayere @LinusMuema @ahmadmardeni1
Why's no one reviewing this article?
Is there a problem with it

**Comment by hectorkambow** [comment]
@Mirror98 
Good afternoon and thank you for submitting your article for publishing. 
With the recent increase in article volume out team has been working diligently to review every article. Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Comment by Mirror98** [comment]
@hectorkambow there is no peer reviewer for the article pull request at the moment

**Comment by hectorkambow** [comment]
@section-engineering-education/peer-reviewers

is anyone available to jump on this PR ASAP.

Thank you

**Comment by hectorkambow** [comment]
@section-engineering-education/content-moderator-team 

Is anyone available to help start a review on this PR. 

I'm happy to get it started - but I will need someone to help with the tech review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@Mirror98 

Please review the edits I've made so far - there were various sentences that were a bit difficult to understand. 

Our EngEd program does not provide revision services - so we ask students to submit polished (and ready to publish) articles to help speed up any review process and to get the content published asap.

Pull requests that are submitted (unpolished) will be closed to allow the author the time to make any needed edits. 

Please go through my edits so far to see if the article can be further polished as my clean up was rather hi-level. 

This PR just needs a tech review and then is should be ready to go.  

cc @section-engineering-education/content-moderator-team

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+---
+Over the years, data has been an essential part of how the world works. These data range from GDP to blood samples to every aspect of the world. As data grew, statistics found ways to extract meaning from them. 
+<!--more-->
+One of these methods is the analysis of variance (ANOVA). ANOVA is a set of statistical models that analyses the differences among means. Microsoft came up with statistical add-ins within Excel to solve some of these problems. Although Microsoft Excel is still rather limited. Later came tools like R-studio and Python. 
+
+With Python, the test for normality and homogeneity became easier. This tutorial will focus on creating an ANOVA table using Microsoft Excel, box plots to test the normality of a one way ANOVA, and Bartlett's rule for homogeneity with Python.
+
+### Prerequisites
+This tutorial requires readers to have proper knowledge on the following:
+- Here is a link to the [Analysis of variance (ANOVA)](https://www.investopedia.com/terms/a/anova.asp) documentation.
```
Comment (Bayler) [inline content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:25]: - Analysis of variance (ANOVA). Here's a [link]((https://www.investopedia.com/terms/a/anova.asp)) to the documentation.

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+- Open our terminal (either our ide's terminal or command prompt).
+
+>Note: If you are using the command prompt, make sure to add the default folder to PATH.
+
+- Type in the following command:
+```bash
+python -m pip install pandas
+```
+
+Wait a moment till the necessary dependencies are installed.
```
Comment (Bayler) [inline content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:57]:
```suggestion
Wait a moment till the dependency is installed. 
```

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+### Getting started
+#### Creating an ANOVA table with Excel
+Microsoft Excels's Analysis Toolpak add-in is a game-changer in statistical Analysis. We'll show you how to create an ANOVA table containing the sum of squares, degree of freedom, mean squares, F value, P-value, F critical.
+
+To perform a one way ANOVA, implement the following step.
+- Import your data set in any preferred Excel format.
+
+![data set](/engineering-education/Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python/data-set.jpg)
+
```
Comment (Bayler) [inline content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:78]: This is not the correct link for the image. You need to confirm the correct link from your files

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+Pandas is used to import a wide variety of data formats, while Matplotlib and Seaborn are visualisation libraries used to generate powerful visualisations. Seaborn is a library that compiles statistical graphics in Python.
+
+It uses the matplotlib library. Next, we'll read our dataset (it will be in CSV file format). Then we will reshape the data frame suitable for the statistical model package and replace the column names. Now we'll generate a boxplot to see the data distribution by treatments.
+
+>Note: When reading from a directory, use double backslash or put `r` at the front of the directory address encased in a quote symbol.
+
+![unicode error](/engineering-education/Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python/unicode-error.jpg)
+
+Below is an implementation of the above instructions:
+```Python
```
Comment (Bayler) [inline content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:114]: Please change the 'P' in Python to be in small letters. Also, change it in the code snippet below.

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+data_set = pd.read_csv(r"C:\Users\DELL\Desktop\Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python\assignments.csv")
+data_set_melt = pd.melt(data_set.reset_index(), id_vars=['index'], value_vars=['A', 'B', 'C', 'D', 'E'])
+data_set_melt.columns = ['index', 'treatments', 'value']
+ax = sns.boxplot(x='treatments', y='value', data=data_set_melt, color='#99c2a2')
+ax = sns.swarmplot(x="treatments", y="value", data=data_set_melt, color='#7d0013')
+plt.show()
```
Comment (Bayler) [inline content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:123]: What's happening here in this code snippet? It's not enough to just write the code. You need to explain clearly to the readers.

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+```
+
+On implementation, our box plot will be shown as below:
+
+![boxplot](/engineering-education/Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python/boxplot.jpg)
+
+#### Testing for homogeneity using Bartlett's rule
+To do this, we'll import scipy.stats and call bartlett's method on our data set.
```
Comment (Bayler) [inline content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:131]: In a paragraph or two, you can begin by explaining to our readers what the Bartlett's rule entail before writing the code.

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+D = 11,18,19,19,15
+E = 9,18,19,23,11
+import scipy.stats as stats 
+print(stats.bartlett(A, B, C, D, E))
+```
+
+When we run the above code on our terminal, we should get:
+
+![bartlett](/engineering-education/Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python/bartlett.jpg)
+
```
Comment (Bayler) [inline content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:147]: Just like the other ones. Make sure your links are correct. And, no CAPS inside the link. Only small letters

**Comment by Bayler** [comment]
@Mirror98 Please work on the few revisions to fine tune your tutorial before we can move on for final review.

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md (hunk 4 above)
Comment (Mirror98) [reply content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:113]: A content moderator @lalith1403 told me that I should make the first p in Python capital.

File: content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md (hunk 5 above)
Comment (Mirror98) [reply content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md:122]: I made an explanation for the code snippet below in the text above.

**Comment by Mirror98** [comment]
@Bayler i've implemented the necessary changes

**Comment by Bayler** [comment]
> @Bayler i've implemented the necessary changes 

I'll check it out

**Comment by hectorkambow** [comment]
> > @Bayler i've implemented the necessary changes
> 
> I'll check it out

Lets keep the back and forth to a min - if major changes are needed lets point that out. Authors are responsible to go through the entire content to find any other errors. 👍

**Comment by hectorkambow** [comment]
@Bayler can we get a status on this PR - just trying to clear our the queue where possible.

**Comment by Bayler** [comment]
> @Bayler can we get a status on this PR - just trying to clear our the queue where possible.

@hectorkambow It's now ready for final review.

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**